src/Interpolation1D.jl

```diff
@@ -86,22 +90,58 @@ function interpolate1d!(
             colidx = Tuple(colcidx)
             colidxsource = _get_grid_colidx(NSG, colidx)
             colidxtarget = _get_grid_colidx(NTG, colidx)
-            first = 1
-            for i1 in 1:ntarget
-                (st, en) = get_stencil(
-                    order,
-                    view(xsource, :, colidxsource...),
-                    xtarget[i1, colidxtarget...],
-                    first = first,
-                    extrapolate = extrapolate,
-                )
-                first = st
-                ftarget[i1, colidx...] = interpolate(
-                    view(xsource, st:en, colidxsource...),
-                    view(fsource, st:en, colidx...),
-                    xtarget[i1, colidxtarget...],
-                )
-            end
+            interpolate_column!(
+                view(ftarget, :, colidx...),
+                view(xsource, :, colidxsource...),
+                view(xtarget, :, colidxtarget...),
+                view(fsource, :, colidx...),
+                order,
+                extrapolate,
+                reverse = reverse,
+            )
+        end
+    end
+    return nothing
+end
+
+"""
+    interpolate_column!(
+        ftarget,
+        xsource,
+        xtarget,
+        fsource,
+        order,
+        extrapolate;
+        reverse = false,
+    )
+
+Interpolate `fsource`, defined on column (`1D`) grid `xsource`, onto the `xtarget` grid.
+It is assumed that both `xsource` and `xtarget` are either mononically increasing 
+(`reverse = false`) or decreasing (`reverse = true`).
+"""
+@inline function interpolate_column!(
```

Comment:
Looks like this is an internal function ? Maybe add a comment in the docstring to differentiate between `interpolate1d!` and `interpolate_column` usage